sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -305,6 +305,19 @@ class HiveDDLSuite
     }
   }
 
+  private def checkMisuseForAlterTableOrView(sqlText: String, isAlterView: Boolean): Unit = {
```

Comment:
I'd like to create a `assertErrorForAlterTableOnView`, which only checks ALTER TABLE when the table is a view. For ALTER VIEW when table is not a view, we can create a new method, or just duplicate the code if the test cases are not many.